source/common/upstream/load_balancer_impl.cc

```diff
@@ -783,6 +783,19 @@ EdfLoadBalancerBase::EdfLoadBalancerBase(
           recalculateHostsInSlowStart(hosts_added);
         }
       });
+
+  min_weight_percent_ = 0.1;
+  if (slow_start_config.has_value()) {
+    if (slow_start_config.value().min_weight_percent() > 0.0 &&
+        slow_start_config.value().min_weight_percent() < 100.0) {
```

Comment:
what if remove the PGV validator for this field, just use the constructor to init the value. It will throw an exception if the PGV failed, which cause the cluster can't be adding to manager. I think the `min_weight_percent` should not affect the cluster update, so just the warning message and use the default value 0.1 is enough ? 

Also the `min_weight_percent` can't be 0 which will cause a really big edf deadline, and we want use the default value if the user not specify a value. so the constraint in proto use `gte: 0.0`(pb default value 0.0) and the constructor use `> 0.0`.

Reply:
PGV is meant to detect problems early and reject configs that are incorrect, so I encourage using that.

If `min_weight_percent` must be greater than 0.0, why not use `gt` instead of `gte`? Same goes for the upper bound (see [here](https://github.com/envoyproxy/protoc-gen-validate#numerics) for more info)

Reply:
PING?

Reply:
sorry, these days i am in a vacation. i will use `{lt: 1.0 gt: 0.0 ignore_empty: true}` to the field `min_weight_percent`, which allows users has not to specify the default value. and will keep checking logic in constructor to ensure the default value to 0.1.